pkg/aws/eni/limits/limits.go

```diff
@@ -867,19 +128,68 @@ func UpdateFromEC2API(ctx context.Context, api ec2API) error {
 	return nil
 }
 
-// parseLimitString returns the Limits struct parsed from config string.
-func parseLimitString(limitString string) (limit ipamTypes.Limits, err error) {
-	intSlice := make([]int, 3)
-	stringSlice := strings.Split(strings.ReplaceAll(limitString, " ", ""), ",")
-	if len(stringSlice) != 3 {
-		return limit, fmt.Errorf("invalid limit value")
+func (l *LimitsGetter) newLimitsUpdateTrigger(api ec2API, minInterval, timeout time.Duration, retryCount int) (*trigger.Trigger, error) {
+	return trigger.NewTrigger(trigger.Parameters{
+		Name:        "ec2-instance-limits-update",
+		MinInterval: minInterval,
+		TriggerFunc: func(reasons []string) {
+			if err := l.updateFromEC2APIWithRetry(context.TODO(), api, timeout, retryCount); err != nil {
+				log.WithError(err).Warning("Failed to update instance limits from EC2 API")
+			}
+
+			l.Lock()
+			l.lastUpdate = time.Now()
+			oldDone := l.triggerDone
+			l.triggerDone = make(chan struct{})
+			l.Unlock()
+			if oldDone != nil {
+				close(oldDone)
+			}
+		},
+	})
+}
+
+func (l *LimitsGetter) initEC2APIUpdateTrigger(api ec2API, triggerMinInterval, ec2APITimeout time.Duration, ec2APIRetryCount int) error {
+	l.Lock()
+	l.triggerMinInterval = triggerMinInterval
+	l.ec2APITimeout = ec2APITimeout
+	l.ec2APIRetryCount = ec2APIRetryCount
+	l.Unlock()
```

Comment:
nit (optional): `newLimitsUpdateTrigger` shouldn't actually run the trigger func. So this `Unlock` here is technically not necessary, we could keep the lock while calling `newLimitsUpdateTrigger` without any danger of deadlocks.

But having said that, I'm fine to keep the code as is.